src/com/google/enterprise/adaptor/fs/FsAdaptor.java

```diff
@@ -608,23 +625,22 @@ private void validateStartPath(Path startPath, boolean logging)
              + " is not valid path - " + e.getMessage() + ".");
     }
 
+    // Do this as soon as possible, since it is selective in how it handles
+    // various exceptions.
+    validateShare(startPath);
+
     if (!crawlHiddenFiles && delegate.isHidden(startPath)) {
       throw new InvalidConfigurationException("The path " + startPath + " is "
           + "hidden. To crawl hidden content, you must set the configuration "
           + "property \"filesystemadaptor.crawlHiddenFiles\" to \"true\".");
     }
 
-    // TODO(mifern): Using a path of \\host\ns\link\FolderA will be
-    // considered non-DFS even though \\host\ns\link is a DFS link path.
-    // This is OK for now since it will fail all three checks below and
-    // will throw an InvalidConfigurationException.
     if (delegate.isDfsLink(startPath)) {
       Path dfsActiveStorage = delegate.resolveDfsLink(startPath);
       if (logging) {
         log.log(Level.INFO, "Using a DFS path resolved to {0}",
                 dfsActiveStorage);
       }
-      validateShare(startPath);
     } else if (delegate.isDfsNamespace(startPath)) {
       if (logging) {
         log.log(Level.INFO, "Using a DFS namespace {0}", startPath);
```

Comment:
I can't comment two lines down, where enumerateDfsLinks is called. This is still always called here, whether allowFiles is true or false, unlike the code in validateShare.